study-builder/tenants/register-user-in-pepper.js

```diff
@@ -90,11 +90,21 @@ function (user, context, callback) {
         } else if (context.request.body.temp_user_guid) {
             pepper_params.tempUserGuid = context.request.body.temp_user_guid;
             console.log('Temp user guid passed in (via body) = ' + pepper_params.tempUserGuid);
-        } else if (user.user_metadata.temp_user_guid) {
-            pepper_params.tempUserGuid = user.user_metadata.temp_user_guid;
-            console.log('No temp user guid found in request, taking one from user metadata');
         } else {
-            console.log('No temp user guid found in request nor user metadata');
+            console.log('No temp user guid passed in request');
+        }
+
+        /**
+         * If `tempUserGuid` was not set with value from request
+         * AND user is not yet registered in pepper (`user.app_metadata.user_guid` is empty)
+         * take `tempUserGuid` from `user_metadata` (if one exists)
+         */
+        if (
+            !pepper_params.tempUserGuid &&
+            !user.app_metadata.user_guid &&
+            !!user.user_metadata.temp_user_guid
```

Comment:
need to delete !! in !!user.user_metadata.temp_user_guid

Reply:
I don't think so- the `!!` is coercing `user.user_metadata.temp_user_guid` into a boolean type.

Reply:
It may look cleaner, but removing the `!!` also makes the statement less safe.

Reply:
Would that decrease the safety of the statement? My understanding of the `!!` is it forces the value to a boolean type, and prevents some type safety issues from cropping up.